pkg/api/types.go

```diff
@@ -708,12 +711,27 @@ type V20180331ARMManagedContainerService struct {
 	*v20180331.ManagedCluster
 }
 
+// AzureStackMetadata is the type for Azure Stack metadata endpoints
+type AzureStackMetadataEndpoints struct {
+	GalleryEndpoint string                            `json:"galleryEndpoint,omitempty"`
+	GraphEndpoint   string                            `json:"graphEndpoint,omitempty"`
+	PortalEndpoint  string                            `json:"portalEndpoint,omitempty"`
+	Authentication  *AzureStackMetadataAuthentication `json:"authentication,omitempty"`
+}
+
+// AzureStackMetadataAuthentication is the type for Azure Stack metadata authentication endpoints
+type AzureStackMetadataAuthentication struct {
+	LoginEndpoint string   `json:"loginEndpoint,omitempty"`
```

Comment:
I think the code base has a naming convention for properties that represent URLs or Base URLs.
In this case it would be `LoginEndpointURL` or `LoginEndpointURLBase` depending on the value.

@CecileRobertMichon, may be able to confirm/deny?

Reply:
No strong opinion here, I'll leave it up to you. If the expected is a full URL then `URL` suffix makes sense. `URLBase` should only be if we expect a suffix and this is only the base domain (eg. `gcr.io`).